site/source/tutorials/1020_registries/index.md

```diff
@@ -0,0 +1,130 @@
+---
+
+{% section 'first' %}
+
+# Working with the Registry
+
+## Overview
+
+Flexibility to override the default type of a widget's children provides a powerful configuration option when it comes to using and customizing widgets with any web application. Additionally as web applications grow, the physical size of the resources required to load the application becomes more and more critical. Keeping the size of resources required to load a web application as small as possible ensures that the application can provide an optimal performance for all users. To help with these challenges, Dojo 2 provides a concept of a `registry` that can be used to achieve both of these goals in a simple and effective manner without intruding on the existing development experience.
+
+In this tutorial, we will start with an application that is that uses concrete widget classes and loads all required assets when the application initially. First we'll swap all the concrete widget references to load the widgets from a `registry`. Then create a new widget that will be lazily loaded when the worker card is clicked the first time.
+
+## Prerequisites
+You can [download](../assets/1020_registries-initial.zip) the demo project and run `npm install` to get started.
+
+The `@dojo/cli` command line tool should be installed globally. Refer to the [Dojo 2 local installation](../000_local_installation/) article for more information.
+
+You also need to be familiar with TypeScript as Dojo 2 uses it extensively. For more information, refer to the [TypeScript and Dojo 2](../comingsoon.html) article.
+
+{% section %}
+
+## The Default Registry
+
+{% task 'Create a default registry.' %}
+
+The first step is to create a `registry` that will be made available to the application by passing the instance as the `registry` property on the `projector`.
+
+{% instruction 'Add the `Registry` import the `main.ts` module.' %}
+
+{% include_codefile 'demo/finished/biz-e-corp/src/main.ts' line:3 %}
+
+{% instruction 'Now, create a `Registry` instance.' %}
+
+{% include_codefile 'demo/finished/biz-e-corp/src/main.ts' line:17 %}
+
+{% instruction 'And finally set the `registry` on the `projector`.' %}
+
+{% include_codefile 'demo/finished/biz-e-corp/src/main.ts' line:30 %}
+
+{% aside 'Registries Everywhere!' %}
+A `registry` can also be used to define widgets with non visual responsibilities, such as state injection and routing. To learn more, take a look at the [container tutorial](../comingsoon.html) and [routing tutorial](../comingsoon.html) .
+{% endaside %}
+
+At the moment we haven't affected the application, however we now have a handle to a `registry` that we can start to define widgets. Once the widgets are defined in the `registry` they will be available through the application and can be used by switching the concrete class in `w()` with the `registry` label.
+
+{% instruction 'Add the widget imports to `main.ts`.' %}
```

Comment:
Should we note that it is best-practice to prefix third party widgets, such as the `dojo` in `dojo-text-input` or do we think this is up to each developer? If we think this is best-practice, might be good to add an aside that discusses it.

Reply:
I don't know if there is truly a best practice, I just did this in the example...